apps/gdalalg_raster_proximity.cpp

```diff
@@ -0,0 +1,199 @@
+
+#include "gdalalg_raster_proximity.h"
+
+#include "cpl_conv.h"
+#include "cpl_vsi_virtual.h"
+
+#include "gdal_alg.h"
+#include "gdal_priv.h"
+#include "gdal_utils.h"
+#include "commonutils.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*      GDALRasterProximityAlgorithm::GDALRasterProximityAlgorithm()            */
+/************************************************************************/
+
+GDALRasterProximityAlgorithm::GDALRasterProximityAlgorithm()
+    : GDALAlgorithm(NAME, DESCRIPTION, HELP_URL)
+{
+
+    AddProgressArg();
+
+    AddOpenOptionsArg(&m_openOptions);
+    AddInputFormatsArg(&m_inputFormats)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES, {GDAL_DCAP_RASTER});
+    AddInputDatasetArg(&m_inputDataset, GDAL_OF_RASTER);
+
+    AddOutputDatasetArg(&m_outputDataset, GDAL_OF_RASTER);
+    AddOutputFormatArg(&m_outputFormat, /* bStreamAllowed = */ false,
+                       /* bGDALGAllowed = */ false)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES,
+                         {GDAL_DCAP_RASTER, GDAL_DCAP_CREATE});
+    AddCreationOptionsArg(&m_creationOptions);
+    AddOverwriteArg(&m_overwrite);
+    AddOutputDataTypeArg(&m_outputDataType)
+        .SetChoices("Byte", "UInt16", "Int16", "UInt32", "Int32", "Float32",
+                    "Float64")
+        .SetDefault(m_outputDataType);
+    AddBandArg(&m_inputBand);
+    AddArg("target-values", 0, _("Target pixel values"), &m_targetPixelValues);
+    AddArg("distance-units", 0, _("Distance units"), &m_distanceUnits)
+        .SetChoices("pixel", "geo")
+        .SetDefault(m_distanceUnits);
+    AddArg("max-distance", 0,
+           _("Maximum distance. The nodata value will be used for pixels "
+             "beyond this distance"),
+           &m_maxDistance)
+        .SetDefault(m_maxDistance);
+    AddArg("fixed-buffer", 0,
+           _("Fixed buffer value (instead of the actual distance)"),
+           &m_fixedBufferValue)
+        .SetMinValueIncluded(0)
+        .SetDefault(m_fixedBufferValue);
+    AddArg("nodata", 0,
+           _("Specify a nodata value to use for pixels that are beyond the "
+             "maximum distance"),
+           &m_noDataValue);
+}
+
+/************************************************************************/
+/*                 GDALRasterProximityAlgorithm::RunImpl()              */
+/************************************************************************/
+
+bool GDALRasterProximityAlgorithm::RunImpl(GDALProgressFunc pfnProgress,
+                                           void *pProgressData)
+{
+
+    auto srcDS = m_inputDataset.GetDatasetRef();
+    CPLAssert(srcDS);
+
+    const auto srcBand = srcDS->GetRasterBand(m_inputBand);
+    // Check done by GDALAlgorithm::AddBandArg()
+    CPLAssert(srcBand);
+
+    auto dstDs = m_outputDataset.GetDatasetRef();
+
+    if (!dstDs && !m_outputDataset.GetName().empty())
```

Comment:
Testing dstDs is useless here as it cannot be non-null.
And you shouldn't test for !m_outputDataset.GetName().empty(), as an empty name is valid for a MEM dataset

Reply:
removed